File: lib/get_ch_basis.py
import numpy as np


def get_ch_basis(th_ax, order, angle=None):
    if angle is not None:
        B = np.zeros((len(th_ax), order + 1))
        for n in range(order + 1):
            B[:, n] = np.cos(n * (th_ax - angle))
    else:
        Acos = np.zeros((len(th_ax), order + 1))
        Asin = np.zeros((len(th_ax), order + 1))
        for n in range(order + 1):
            Acos[:, n] = np.cos(n * th_ax)
            Asin[:, n] = np.sin(n * th_ax)
        B = np.hstack([Acos, Asin])

    return B

File: parametric_vm/lib/getchbasis.m
function [ B ] = getchbasis(th_ax,order,angle)
%GETCHBASIS Summary of this function goes here
%   Detailed explanation goes here
% N = 180
% th_ax=(0:2*pi/N:2*pi*(1-1/N))';
th_ax = double(th_ax);
order = double(order);
angle = double(angle);
if nargin==3
    B=zeros(length(th_ax),order+1);
    for n=0:order
        B(:,n+1)=cos(n*(th_ax-angle));
    end
else
    Acos=zeros(length(th_ax),order+1);
    Asin=zeros(length(th_ax),order+1);
    for n=0:order
       Acos(:,n+1)=cos(n*(th_ax));
       Asin(:,n+1)=sin(n*(th_ax));
    end 
    B=[Acos,Asin];
end
end
